numba/cgutils.py

```diff
@@ -656,16 +656,16 @@ def unpack_tuple(builder, tup, count=None):
     return vals
 
 
-def get_item_pointer(builder, aryty, ary, inds, wraparound=False):
+def get_item_pointer(context, builder, aryty, ary, inds, wraparound=False, boundscheck=True):
```

Comment:
I'm not sure that this should be on by default as I think it may prevent a loop containing operations derived from array indexing from being vectorized/unwound etc.

Reply:
Just noted this:
> Add the flag to the public API (right now it is enabled by default for testing purposes

so the comment can probably be ignored :)

Reply:
Yeah, I definitely don't want to change existing compatibility here (or at least I'll let you guys decide if that's what you want to do). I just left this as this for now because I haven't yet figured out how to pass this through as a real option on `jit`.